lms/static/scripts/frontend_apps/utils/dashboard-opening-form-controller.tsx

```diff
@@ -0,0 +1,48 @@
+import { render } from 'preact';
+
+import type { DashboardConfig } from '../config';
+
+/**
+ * Handles the rendering and submitting of the form which is used to open the
+ * instructor dashboard
+ */
+export class DashboardOpeningFormController {
+  private _dashboardConfig: DashboardConfig;
+  private _authToken: string;
+
+  constructor(dashboardConfig: DashboardConfig, authToken: string) {
+    this._dashboardConfig = dashboardConfig;
+    this._authToken = authToken;
+  }
+
+  render() {
+    const container = document.createElement('div');
+    document.body.append(container);
+
+    const destroy = () => {
+      render(null, container);
+      container.remove();
+    };
+
+    render(
+      <form
+        target="_blank"
+        method="POST"
+        action={this._dashboardConfig.dashboardEntryPoint.path}
+        ref={form => {
+          if (!form) {
+            return;
+          }
+
+          // Immediately submit the form, as soon as it has been mounted
+          form.submit();
+          destroy();
```

Comment:
For simplicity, I have decided to add this side effect directly in the form's ref callback, because it only needs to run once, as the form gets rendered.

If at some point this logic becomes more complex, we should extract the rendering of the form to a component, and handle the side effect via `useEffect`.

Reply:
You could do something even simpler, along the lines of:

```js
const container = document.createElement('div');
render(<form .../>, container);
container.querySelector('form').submit()
container.remove()
```

Here the only use of Preact is just a more convenient way to create the form element than constructing it manually, although that would also be pretty simple with basic DOM APIs.